parallel-consumer-core/src/main/java/io/confluent/parallelconsumer/internal/AbstractParallelEoSStreamProcessor.java

```diff
@@ -470,27 +473,50 @@ private void checkAutoCommitIsDisabled(org.apache.kafka.clients.consumer.Consume
-                ConsumerCoordinator coordinator = (ConsumerCoordinator) coordinatorField.get(consumer); //IllegalAccessException
-
-                if (coordinator == null)
-                    throw new IllegalStateException("Coordinator for Consumer is null - missing GroupId? Reflection broken?");
-
-                Field autoCommitEnabledField = coordinator.getClass().getDeclaredField("autoCommitEnabled");
-                autoCommitEnabledField.setAccessible(true);
-                Boolean isAutoCommitEnabled = (Boolean) autoCommitEnabledField.get(coordinator);
+                Field delegateField = KafkaConsumer.class.getDeclaredField("delegate");
+                delegateField.setAccessible(true);
+                ConsumerDelegate<?, ?> delegate = (ConsumerDelegate<?, ?>) delegateField.get(consumer);
+
+                Boolean isAutoCommitEnabled;
+                if (delegate instanceof LegacyKafkaConsumer) {
+                    Field coordinatorField = LegacyKafkaConsumer.class.getDeclaredField("coordinator");
+                    coordinatorField.setAccessible(true);
+                    ConsumerCoordinator coordinator = (ConsumerCoordinator) coordinatorField.get(delegate); //IllegalAccessException
+
+                    if (coordinator == null)
+                        throw new IllegalAccessException("Coordinator for Consumer is null - missing GroupId? Reflection broken?");
+
+                    Field autoCommitEnabledField = coordinator.getClass().getDeclaredField("autoCommitEnabled");
+                    autoCommitEnabledField.setAccessible(true);
+                    isAutoCommitEnabled = (Boolean) autoCommitEnabledField.get(coordinator);
+                } else if (delegate instanceof AsyncKafkaConsumer) {
+                    Field autoCommitEnabledField = AsyncKafkaConsumer.class.getDeclaredField("autoCommitEnabled");
+                    autoCommitEnabledField.setAccessible(true);
+                    isAutoCommitEnabled = (Boolean) autoCommitEnabledField.get(delegate);
+                } else {
+                    log.error("Consumer delegate is neither a LegacyKafkaConsumer nor a AsyncKafkaConsumer - cannot check auto commit is disabled for consumer type: " + consumer.getClass().getName());
+                    isAutoCommitEnabled = false;
+                }
 
-                if (TRUE.equals(isAutoCommitEnabled))
-                    throw new ParallelConsumerException("Consumer auto commit must be disabled, as commits are handled by the library.");
+                if (TRUE.equals(isAutoCommitEnabled)) {
+                    if (options.isSoftAutoCommitDisabledCheck()) {
+                        log.warn("Consumer auto commit must be disabled, as commits are handled by the library.");
+                    } else {
+                        throw new ParallelConsumerException("Consumer auto commit must be disabled, as commits are handled by the library.");
+                    }
+                }
```

Comment:
Revert this check back - only disabling error on reflective access - if auto commit field is accessible / found - and is not disabled - shut down with an error.
```suggestion
                if (TRUE.equals(isAutoCommitEnabled))
                    throw new ParallelConsumerException("Consumer auto commit must be disabled, as commits are handled by the library.");
```